tools/clang/lib/Sema/SemaHLSLDiagnoseTU.cpp

```diff
@@ -475,6 +477,18 @@ void hlsl::DiagnoseTranslationUnit(clang::Sema *self) {
       HLSLMethodCallDiagnoseVisitor Visitor(self, shaderModel, EntrySK, FDecl,
                                             DiagnosedCalls);
       Visitor.TraverseDecl(FD);
+
+      // diagnose any node functions that have incompatible launch types with
+      // the present SV semantics on each parameter.
+      if (EntrySK == DXIL::ShaderKind::Node) {
+        if (const auto *NodeLaunchAttr =
+                FDecl->getAttr<clang::HLSLNodeLaunchAttr>()) {
```

Comment:
Question: Can we not assume that there is a launch attr for a node shader at this point? Do we catch that?

Reply:
We don't catch it before this point, and we cannot assume there will be a launch attr.
One example of this is tools\clang\test\SemaHLSL\shader_attribute_no_mismatch.hlsl:
```
[shader("node")] 
[shader("node")] 
[nodedispatchgrid(8,1,1)]
[ numthreads( 64, 2, 2 ) ]  /* expected-no-diagnostics */
void NNMain() {
}
```
which will cause an ```Internal compiler error: access violation. Attempted to read from address 0x0000000000000010```
So I must leave the check here.